src/components/CodeAndPreview.tsx

```diff
@@ -41,12 +42,22 @@ export function CodeAndPreview({ snippet }: Props) {
     )
   }, [])
 
-  // Initialize with template or snippet's manual edits if available
-  const [manualEditsFileContent, setManualEditsFileContent] = useState(
-    snippet?.manual_edits_json ??
-      JSON.stringify(manualEditsTemplate, null, 2) ??
-      "",
-  )
+  // Initialize manualEditsFileContent with proper validation
+  const [manualEditsFileContent, setManualEditsFileContent] = useState(() => {
+    try {
+      const initialContent =
+        snippet?.manual_edits_json ??
+        JSON.stringify(manualEditsTemplate, null, 2)
+      // Validate that it's parseable JSON
+      JSON.parse(initialContent)
+      return initialContent
+    } catch (e) {
+      console.warn(
+        "Invalid initial manual edits content, using default template",
+      )
+      return JSON.stringify(manualEditsTemplate, null, 2)
+    }
```

Comment:
don't have redundant logic for the default manual edits json, the `useEffect` already handles it